overlord/snapstate/handlers_aliasesv2_test.go

```diff
@@ -671,95 +1323,292 @@ func (s *snapmgrTestSuite) TestDoUndoSetupAliasesAuto(c *C) {
 	c.Assert(err, IsNil)
 
 	c.Check(snapst.AutoAliasesDisabled, Equals, false)
-	c.Check(snapst.AliasesPending, Equals, true)
+	c.Check(snapst.AliasesPending, Equals, false)
+	c.Check(snapst.Aliases, DeepEquals, map[string]*snapstate.AliasTarget{
+		"alias1": {Auto: "cmd1"},
+		"alias2": {Auto: "cmd2x"},
+		"alias3": {Auto: "cmd3"},
+	})
 }
 
-func (s *snapmgrTestSuite) TestDoSetupAliasesNothing(c *C) {
+func (s *snapmgrTestSuite) TestDoUndoSetupAliasesAutoErrorMidwayPruneOldAliases(c *C) {
 	s.state.Lock()
 	defer s.state.Unlock()
 
+	snapstate.AutoAliases = func(st *state.State, info *snap.Info) (map[string]string, error) {
+		c.Check(info.InstanceName(), Equals, "alias-snap")
+		return map[string]string{
+			"alias1": "cmd1",
+			"alias2": "cmd2",
+			"alias4": "cmd4",
+		}, nil
+	}
+
 	snapstate.Set(s.state, "alias-snap", &snapstate.SnapState{
 		Sequence: snapstatetest.NewSequenceFromSnapSideInfos([]*snap.SideInfo{
 			{RealName: "alias-snap", Revision: snap.R(11)},
 		}),
-		Current: snap.R(11),
-		Active:  true,
+		Current:             snap.R(11),
+		Active:              true,
+		AutoAliasesDisabled: false,
+		AliasesPending:      true,
```

Comment:
this is strange as initial state for this?  initially disk and state match, no?  please double check that everywhere we set AliasesPending in a fixture it is matches expected reality